controllers/nstemplateset/nstemplatetier.go

```diff
@@ -11,37 +11,48 @@ import (
 	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
 	"github.com/codeready-toolchain/toolchain-common/pkg/template"
 	templatev1 "github.com/openshift/api/template/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
-
 	"github.com/pkg/errors"
+	errs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var tierTemplatesCache = newTierTemplateCache()
 
-// getTierTemplate retrieves the TierTemplate resource with the given name from the host cluster
+// getTierTemplate retrieves the TierTemplateRevision resource with the given name from the host cluster,
+// if not found then falls back to the current logic of retrieving the TierTemplate
 // and returns an instance of the tierTemplate type for it whose template content can be parsable.
 // The returned tierTemplate contains all data from TierTemplate including its name.
 func getTierTemplate(ctx context.Context, hostClusterFunc cluster.GetHostClusterFunc, templateRef string) (*tierTemplate, error) {
+	var tmpl *toolchainv1alpha1.TierTemplate
 	if templateRef == "" {
-		return nil, fmt.Errorf("templateRef is not provided - it's not possible to fetch related TierTemplate resource")
+		return nil, fmt.Errorf("templateRef is not provided - it's not possible to fetch related TierTemplate/TierTemplateRevision resource")
 	}
+
 	if tierTmpl, ok := tierTemplatesCache.get(templateRef); ok && tierTmpl != nil {
 		return tierTmpl, nil
 	}
-	tmpl, err := getToolchainTierTemplate(ctx, hostClusterFunc, templateRef)
+	ttr, err := getToolchainTierTemplateRevision(ctx, hostClusterFunc, templateRef)
 	if err != nil {
-		return nil, err
+		if errs.IsNotFound(err) {
+			tmpl, err = getToolchainTierTemplate(ctx, hostClusterFunc, templateRef)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
 	}
 	tierTmpl := &tierTemplate{
 		templateRef: templateRef,
 		tierName:    tmpl.Spec.TierName,
```

Comment:
This will fail if the `ttr` is found, because `tmpl` will be `nil` in that case.

Reply:
+1

we'll also need to add a unit test verifying that `getTierTemplate` works when ttr is present.